cmake/kokkos_arch.cmake

```diff
@@ -845,14 +845,35 @@ IF (KOKKOS_ENABLE_OPENACC)
       Clang -Xopenmp-target=nvptx64-nvidia-cuda -march=${CLANG_CUDA_ARCH}
             -fopenmp-targets=nvptx64-nvidia-cuda
     )
+    IF(DEFINED ENV{CUDA_PATH})
+      COMPILER_SPECIFIC_LINK_OPTIONS(Clang -L$ENV{CUDA_PATH}/lib64)
+    ENDIF()
+    COMPILER_SPECIFIC_LIBS(
+      Clang -lcudart
+      NVHPC -cuda
+    )
   ELSEIF(KOKKOS_AMDGPU_ARCH_FLAG)
     COMPILER_SPECIFIC_FLAGS(
       Clang -Xopenmp-target=amdgcn-amd-amdhsa -march=${KOKKOS_AMDGPU_ARCH_FLAG}
             -fopenmp-targets=amdgcn-amd-amdhsa
     )
+    IF(DEFINED ENV{ROCM_PATH})
+      COMPILER_SPECIFIC_FLAGS(Clang -I$ENV{ROCM_PATH}/include)
+      COMPILER_SPECIFIC_LINK_OPTIONS(Clang -L$ENV{ROCM_PATH}/lib)
+    ENDIF()
+    COMPILER_SPECIFIC_LIBS(Clang -lamdhip64)
   ELSE()
+    # When not compiling for offload to any GPU, we're compiling for kernel
+    # execution on the host.  In that case, memory is shared between the OpenACC
+    # space and the host space.
```

Comment:
I thought that this else clause meant that we may run anywhere?
Did I misunderstand?
If not should we then distinguish between forcing to run on host and the generic case?

Reply:
The intention is to follow the standard Kokkos style (decide the target device at compilation time).
If the target GPU architecture is explicitly specified (e.g., -DKokkos_ARCH_HOPPER90=ON), the Openacc backend will offload Kokkos parallel constructs to the target device. (Users can still offload the parallel constructs to the host CPU by inserting the `acc_set_device_type()` API in the application, but `acc_set_device_type() API is not Kokkos API and thus not portable.
If the target GPU architecture is NOT specified, the OpenACC backend will offload Kokkos parallel constructs to the host CPU as a multicore target.

Reply:
Does that mean you want to drop the logic that allows us to fallback to host?
https://github.com/kokkos/kokkos/blob/195f45f8f1c641f551bff791044eaf99c709a029/core/src/OpenACC/Kokkos_OpenACC.cpp#L49-L59

You could very well preserve that logic and just introduce a new OpenACC-only option to force using the host.

Reply:
It will be good to have a new OpenACC-only option to force using the host; @dalg24 can you suggest a good Kokkos CMake option name for that? (I'm not familiar with the Kokkos CMake option naming rule).

Reply:
What about `Kokkos_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE`?

Reply:
Updated cmake/make configurations to use `Kokkos_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE`